spec/models/document_spec.rb

```diff
@@ -833,4 +833,34 @@ def initialize(params = {})
       expect(AaibReport.title).to eq("AAIB Report")
     end
   end
+
+  describe "#send_email_on_publish?" do
+    it "should send email if document is major update and no disable_email_alert override exists" do
+      doc = MyDocumentType.new(update_type: "major")
+      expect(doc.send_email_on_publish?).to be true
+    end
+
+    it "should send email if document is major update and disable_email_alert override is false" do
+      doc = MyDocumentType.new(update_type: "major")
+      doc.disable_email_alert = false
+      expect(doc.send_email_on_publish?).to be true
+    end
+
+    it "(Caution) will stop sending email if document is major update and disable_email_alert override is a random string" do
+      doc = MyDocumentType.new(update_type: "major")
+      doc.disable_email_alert = "invalid"
+      expect(doc.send_email_on_publish?).to be false
+    end
+
+    it "should stop send email if document is major update and disable_email_alert override is true" do
+      doc = MyDocumentType.new(update_type: "major")
+      doc.disable_email_alert = true
+      expect(doc.send_email_on_publish?).to be false
+    end
+
+    it "should stop send email if document is minor update" do
```

Comment:
Is it worth adding a test to ensure that `update_type: minor` and `disable_email_alert: false` doesn't send an email?
(It wouldn't, but someone could probably swap the expressions in the `send_email_on_publish?` method and it might then send).

Reply:
good point - i'll add that

Reply:
Have added a test for this